spec/services/search/book_versions/index_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'rails_helper'
+require 'vcr_helper'
+
+# In this test, the book content is drawn from the file fixture (to limit the
+# book to 1 page) & the page content is drawn from cnx archive and recorded thru
+# vcr
+#
+# ElasticSearch must be running for this test to succeed
+# e.g  docker run
+#          -p 9200:9200 -p 9300:9300
+#          -v elasticsearch:/usr/share/elasticsearch/data
+#          -e "discovery.type=single-node"
+#          docker.elastic.co/elasticsearch/elasticsearch:6.3.2
+RSpec.describe Search::BookVersions::Index, type: :external, vcr: VCR_OPTS do
+  let(:cnx_book_id) { '14fb4ad7-39a1-4eee-ab6e-3ef2482e3e22' }
+  let(:book_version) { "15.1" }
+  let(:test_book_json) { JSON.parse(file_fixture('mini.json').read) }
+
+  subject(:index) { described_class.new(book_guid: cnx_book_id, book_version: book_version) }
+
+  def delete_index
+    if ElasticsearchClient.instance.indices.exists? index: index.name
+      ElasticsearchClient.instance.indices.delete index: index.name
+    end
+  end
+
+  before { delete_index }
+  after { delete_index }
+
+  describe "#create" do
+    it 'creates the index' do
+      index.create
+      expect(ElasticsearchClient.instance.indices.exists?(index: index.name)).to be_truthy
+    end
+  end
+
+  describe "#populate" do
+    let(:test_book_url) {
+      "https://archive.cnx.org/contents/#{cnx_book_id}@#{book_version}"
+    }
+    let(:test_page_url) {
+      "#{test_book_url}:ada35081-9ec4-4eb8-98b2-3ce350d5427f@6"
+    }
+
+    before do
+      allow(OpenStax::Cnx::V1).to receive(:fetch).with(test_book_url).and_return(test_book_json)
+      allow(OpenStax::Cnx::V1).to receive(:fetch).with(test_page_url).and_call_original
+    end
+
+    it 'populates the index' do
+      index.create
+      index.populate
+      sleep 1 # a necessary evil to get the real elastic search document count
```

Comment:
Let's only sleep if we are recording the cassette:

```ruby
sleep 1 if VCR.current_cassette.try(:recording?) # to get the real elasticsearch document count
```

Reply:
@jpslav  it only works w/ the sleep 1 in there either recording or not.